src/plugins/intel_gpu/tests/unit/test_cases/xattention_gpu_test.cpp

```diff
@@ -0,0 +1,1273 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <intel_gpu/primitives/activation.hpp>
+#include <intel_gpu/primitives/data.hpp>
+#include <intel_gpu/primitives/eltwise.hpp>
+#include <intel_gpu/primitives/gemm.hpp>
+#include <intel_gpu/primitives/input_layout.hpp>
+#include <intel_gpu/primitives/paged_attention.hpp>
+#include <intel_gpu/primitives/permute.hpp>
+#include <intel_gpu/primitives/reorder.hpp>
+#include <intel_gpu/primitives/reshape.hpp>
+#include <intel_gpu/primitives/softmax.hpp>
+#include <openvino/reference/xattention.hpp>
+
+#include "openvino/runtime/tensor.hpp"
+#include "primitive_inst.h"
+#include "random_generator.hpp"
+#include "test_utils.h"
+
+using namespace cldnn;
+using namespace ov::intel_gpu;
+using namespace ::tests;
+
+enum class XAttentionScoresMode { DISABLED = 0, LAST_TOKEN, SNAPKV };
+
+struct XAttentionSubsequenceDescriptor {
+    int num_tokens;
+    int past_len;
+};
+
+struct XAttentionCacheRotationDescriptor {
+    bool apply_rotation;
+    // configures 2nd dimension of rotation_deltas
+    // if per_block is true, single value is used for all tokens inside the block
+    // otherwise, each token uses an independent value
+    bool per_block;
+};
+
+struct XAttentionManager {
```

Comment:
@WeldonWangwang Since reference has fixed, can you please remove the copy and use reference directly?

Reply:
Removed